src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -973,6 +973,40 @@ def to_json(self) -> dict:
             del out[e]
         return out
 
+    def to_dict(self) -> dict:
+        out: Dict[str, Any] = OrderedDict()
+        out['daemon_type'] = self.daemon_type
+        out['daemon_id'] = self.daemon_id
+        out['hostname'] = self.hostname
+        out['container_id'] = self.container_id
+        out['container_image_id'] = self.container_image_id
+        out['container_image_name'] = self.container_image_name
+        out['container_image_digests'] = self.container_image_digests
+        out['memory_usage'] = self.memory_usage
+        out['memory_request'] = self.memory_request
```

Comment:
Actually no, you can't just copy the `to_json` method and rename it `to_dict`: 

(Diff at the time you copied the method)

![image](https://user-images.githubusercontent.com/2574405/132660377-0300bcb0-4f72-41a6-9c01-f6205752ff45.png)


because over time they will diverge: 

(Diff on current master HEAD)

![image](https://user-images.githubusercontent.com/2574405/132660642-f6f9c0e9-c8b5-4c2c-9394-0f53659f85a5.png)

Copying such a large method is just bad practice. sorry that I haven't noticed this.

Reply:
Was it a wrongly resolved conflict or what? So it's just a matter of adding the missing fields?

@aaSharma14 please have a look at this, please? Thanks!

Reply:
@epuertat, No its not a wrongly resolved commit. At that time we introduced a new ``to_dict`` method. Over the time new fields were added to the ``to_json`` method..but those were not added to the ``to_dict`` method. So, now both these two methods are not in sync with each other as there are different number of fields in both. A quick solution would be add those new fields to ``to_dict`` method as well. Or if we want to refactor the code..then I will need to find a better solution for this.

Reply:
> @epuertat, No its not a wrongly resolved commit. At that time we introduced a new `to_dict` method. Over the time new fields were added to the `to_json` method.

Right. That's how it is. If you want to avoid having to maintain this method every now and then, you have to unify those two again.

Reply:
Thanks @aaSharma14! I remember I already asked this, but isn't there a way to avoid this by reusing the same method? I find quite unusual to have 2 methods that do almost the same (I'd remove the `to_json()` one as it's not JSON (text) what that's returning but a dict with some subfields in JSON format...

Reply:
The reason I avoided calling those methods `to_dict` is, that they actually sometimes don't return a dict:

```
➜  src git:(_check_for_moved_osds) ✗ find * -name '*.py' | grep -v -e rook_client -e tox -e test | xargs egrep 'def.*to_json'
ceph-volume/venv/lib/python3.9/site-packages/pip/_internal/models/direct_url.py:    def to_json(self):
pybind/mgr/cephadm/services/osd.py:    def to_json(self) -> dict:
pybind/mgr/cephadm/configchecks.py:    def to_json(self) -> Dict[str, Any]:
pybind/mgr/cephadm/configchecks.py:    def to_json(self) -> List[Dict[str, str]]:
pybind/mgr/cephadm/upgrade.py:    def to_json(self) -> dict:
pybind/mgr/cephadm/inventory.py:    def to_json(self) -> Dict[str, Any]:
pybind/mgr/orchestrator/module.py:    def to_json_1(obj: Any) -> Any:
pybind/mgr/orchestrator/module.py:    def to_json_n(objs: List) -> List:
pybind/mgr/orchestrator/_interface.py:    def to_json(self) -> dict:
pybind/mgr/orchestrator/_interface.py:    def to_json(self) -> OrderedDict:
pybind/mgr/orchestrator/_interface.py:    def to_json(self) -> dict:
pybind/mgr/orchestrator/_interface.py:    def to_json(self) -> str:
pybind/mgr/progress/module.py:    def to_json(self):
pybind/mgr/progress/module.py:    def to_json(self):
pybind/mgr/rbd_support/schedule.py:    def to_json(self) -> str:
pybind/mgr/rbd_support/task.py:    def to_json(self) -> str:
pybind/mgr/volumes/fs/volume.py:def name_to_json(names):
python-common/ceph/deployment/inventory.py:    def to_json(self):
python-common/ceph/deployment/inventory.py:    def to_json(self):
python-common/ceph/deployment/hostspec.py:    def to_json(self) -> Dict[str, Any]:
python-common/ceph/deployment/drive_group.py:    def to_json(self):
python-common/ceph/deployment/service_spec.py:    def to_json(self) -> str:
python-common/ceph/deployment/service_spec.py:    def to_json(self) -> dict:
python-common/ceph/deployment/service_spec.py:    def to_json(self):
```